src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/DraftReferralDTO.kt

```diff
@@ -49,14 +66,23 @@ data class DraftReferralDTO(
         whenUnavailable = referral.whenUnavailable,
         additionalRiskInformation = referral.additionalRiskInformation,
         maximumEnforceableDays = referral.maximumEnforceableDays,
-        desiredOutcomes = referral.selectedDesiredOutcomes?.groupBy { it.serviceCategoryId }?.toSortedMap()
-          ?.map { (serviceCategoryId, desiredoutcomes) -> SelectedDesiredOutcomesDTO(serviceCategoryId, desiredoutcomes.map { it.desiredOutcomeId }.sorted()) },
+        desiredOutcomes = referral.selectedDesiredOutcomes?.ifEmpty { null }
+          ?.groupBy { it.serviceCategoryId }
+          ?.toSortedMap()
+          ?.map { (serviceCategoryId, desiredOutcomes) ->
+            SelectedDesiredOutcomesDTO(
+              serviceCategoryId,
+              desiredOutcomes.map { it.desiredOutcomeId }.sorted()
+            )
+          },
         serviceUser = ServiceUserDTO.from(referral.serviceUserCRN, referral.serviceUserData),
         serviceProvider = ServiceProviderDTO.from(contract.primeProvider),
         relevantSentenceId = referral.relevantSentenceId,
         // TODO: remove this once cohort referrals changes are complete
-        serviceCategoryId = if (referral.selectedServiceCategories?.isNotEmpty() == true) referral.selectedServiceCategories!!.elementAt(0).id else null,
-        serviceCategoryIds = referral.selectedServiceCategories?.map { it.id }?.sorted(),
+        serviceCategoryId = referral.selectedServiceCategories?.firstOrNull()?.id,
+        serviceCategoryIds = referral.selectedServiceCategories?.ifEmpty { null }
```

Comment:
should this be unit tested? or is it ok to rely on the contracts to do it?

Reply:
i can add a quick unit test

Reply:
done